### PR TITLE
Minigames Page and Minigames

### PR DESCRIPTION
**Pull request by AnibalDVuo**
Added a Minigames page and the following components:
- Virtual Sear
- Cut Guessr
- Conversation Starter
- Ingredient Match

Also modified the Quiz component.

Added atoms Switch and Tooltip, modified Button and Input

Modified the schema for Step and Recipe (made some variables non mandatory)

### CODE REVIEW

File: ff.web/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "mobx": "^6.12.3",
     "mobx-persist-store": "^1.1.5",
     "mobx-react-lite": "^4.0.7",
+    "next": "^14.2.15",
```
Comment (adamG0315) [inline ff.web/package.json:36]: For what are we using Next exactly?

File: ff.web/src/components/atoms/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  disabled?: boolean;
+  size?: 'small' | 'medium';
+}
+
+function Switch({ checked, onChange, disabled = false, size = 'medium' }: SwitchProps) {
+  const rootStyle: React.CSSProperties = {
+    display: 'inline-block',
+    position: 'relative',
+    width: size === 'small' ? '34px' : '58px',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/Switch.tsx:14]: Use the spacing values from the scss vars.

File: ff.web/src/components/atoms/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    zIndex: 1,
+    margin: 0,
+    cursor: 'inherit',
+  };
+
+  const thumbStyle: React.CSSProperties = {
+    position: 'absolute',
+    display: 'block',
+    backgroundColor: '#fff',
+    width: size === 'small' ? '20px' : '26px',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/Switch.tsx:36]: same here

File: ff.web/src/components/atoms/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    height: size === 'small' ? '20px' : '26px',
+    borderRadius: '50%',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/Switch.tsx:38]: For border radius we also have variables

File: ff.web/src/components/atoms/Switch.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    top: '50%',
+    transform: checked 
+      ? `translateX(${size === 'small' ? '14px' : '28px'}) translateY(-50%)`
```
Comment (adamG0315) [inline ff.web/src/components/atoms/Switch.tsx:41]: scss variables

File: ff.web/src/components/atoms/Switch.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      : 'translateY(-50%)',
+    left: size === 'small' ? '0' : '2px',
+    transition: 'transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+    boxShadow: '0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12)',
+  };
+
+  const trackStyle: React.CSSProperties = {
+    backgroundColor: checked ? '#90caf9' : '#000',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/Switch.tsx:49]: Do we have color variables for these? If not add those to the scss tokens.

File: ff.web/src/components/atoms/ToolTip.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+      setAdjustedPosition(newPosition);
+    }
+  }, [isVisible, position]);
+
+  const tooltipStyle: React.CSSProperties = {
+    position: 'absolute',
+    backgroundColor: 'rgba(0, 0, 0, 0.8)',
+    color: 'white',
+    padding: '5px 10px',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/ToolTip.tsx:56]: use spacing variables

File: ff.web/src/components/atoms/ToolTip.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    borderRadius: '4px',
```
Comment (adamG0315) [inline ff.web/src/components/atoms/ToolTip.tsx:57]: vars...

File: ff.web/src/components/molecules/AnimalImage.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    canvas.width = img.width;
+    canvas.height = img.height;
+
+    if (!userGuess) return;
+
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+    ctx.beginPath();
+    ctx.moveTo(userGuess.x * canvas.width / 100, userGuess.y * canvas.height / 100);
+    ctx.lineTo(actualLocation.x * canvas.width / 100, actualLocation.y * canvas.height / 100);
+    ctx.strokeStyle = 'red';
```
Comment (adamG0315) [inline ff.web/src/components/molecules/AnimalImage.tsx:53]: color vars

File: ff.web/src/components/molecules/AnimalImage.tsx [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    ctx.lineWidth = 2;
+    ctx.stroke();
+
+    // Draw blue circle for user guess
+    ctx.beginPath();
+    ctx.arc(userGuess.x * canvas.width / 100, userGuess.y * canvas.height / 100, 5, 0, 2 * Math.PI);
+    ctx.fillStyle = 'blue';
```
Comment (adamG0315) [inline ff.web/src/components/molecules/AnimalImage.tsx:60]: same here

File: ff.web/src/components/molecules/AnimalImage.tsx [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+    ctx.fill();
+
+    // Draw green circle for actual location
+    ctx.beginPath();
+    ctx.arc(actualLocation.x * canvas.width / 100, actualLocation.y * canvas.height / 100, 5, 0, 2 * Math.PI);
+    ctx.fillStyle = 'green';
```
Comment (adamG0315) [inline ff.web/src/components/molecules/AnimalImage.tsx:66]: and here

File: ff.web/src/components/molecules/AnimalImage.tsx [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+        }}
+      />
+      {developerMode && (
+        <div style={{
+          position: 'absolute',
+          top: 0,
+          left: 0,
+          backgroundColor: 'rgba(0, 0, 0, 0.5)',
+          color: 'white',
+          padding: '0.5rem'
```
Comment (adamG0315) [inline ff.web/src/components/molecules/AnimalImage.tsx:127]: spacing vars pls

File: ff.web/src/components/organisms/ConversationStarter.module.scss [hunk 13]
```diff
@@ -0,0 +1,138 @@
+.sneak_peek_body {
+    color: white;
+}
+
+.header {
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    padding-bottom: 20px;
```
Comment (adamG0315) [inline ff.web/src/components/organisms/ConversationStarter.module.scss:10]: in this whole file change the hardcoded values to the ones defined in the scss file

File: ff.web/src/components/organisms/CutGuessr.module.scss [hunk 14]
```diff
@@ -0,0 +1,134 @@
+.sneak_peek_body {
+    color: white;
+}
+
+.header {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    padding-bottom: 20px;
```
Comment (adamG0315) [inline ff.web/src/components/organisms/CutGuessr.module.scss:10]: vars in the whole file

File: ff.web/src/components/organisms/IngredientMatch.module.scss [hunk 15]
```diff
@@ -0,0 +1,138 @@
+.sneak_peek_body {
+    color: white;
+}
+
+.header {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
```
Comment (adamG0315) [inline ff.web/src/components/organisms/IngredientMatch.module.scss:9]: same here

File: ff.web/src/components/organisms/Quiz.module.scss [hunk 16]
```diff
@@ -94,11 +91,14 @@
   color: var(--text-primary);
   border: 1px solid var(--border-color);
   border-radius: var(--radius-8);
-  padding: var(--space-12) var(--space-16);
+  padding: 12px 16px;
```
Comment (adamG0315) [inline ff.web/src/components/organisms/Quiz.module.scss:94]: aren't these of the same value?

File: ff.web/src/components/organisms/VirtualSear.module.scss [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+
+.tooltip_container {
+    display: flex;
+    justify-content: flex-end;
+    align-items: center;
+}
+
+.title {
+    font-size: 24px;
+    padding-bottom: 20px;
```
Comment (adamG0315) [inline ff.web/src/components/organisms/VirtualSear.module.scss:19]: use vars wherever possible

File: ff.web/package.json (hunk 1 above)
Comment (AnibalDVuo) [reply ff.web/package.json:36]: I think it was a leftover from when I was trying out different packages for components, the import shouldn't be necessary in that case

**Pull request merged**